12_glob_lab/globlab.txt

```diff
@@ -5,25 +5,34 @@ https://www.tocode.co.il/bundles/unix1/lessons/globlab
 
   1. All files whose name starts with a lowercase letter
   Answer: 
-
+ls [a-z]*
   2. All files whose name starts with an uppercase letter
   Answer:
-
+ls [A-Z]*
   3. All files whose name ends in an "a", "e", "i", "o" or "u"
   Answer:
 
+  leo@leo-VirtualBox:~/lab$ ls *.[a,e,i,o,u]
+
   4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
   Answer:
+leo@leo-VirtualBox:~/lab$ ls *.[!a,e,i,o,u]
 
   5. All files with extension ".c" that are in a folder named "src"
   Answer:
-
+ls *.c src/
```

Comment:
here you list all the files in you current dir **and** all the files in ```src/``` your glob pattern is missing the files in src/, it should be ```src/*.js```

why this is important? well because shell globs are expended **before** the command is invoked by the shell, and here ls happens to list all files in a give directory, but it's not always the case.. 

consider this:
```
vitali@Tux:~/work/react-hut$ echo "echo is a builtin command that prints it's arguments"
echo is a builtin command that prints it's arguments

vitali@Tux:~/work/react-hut$ echo src
src

vitali@Tux:~/work/react-hut$ echo src/*.js
src/class-lists.js src/hut.js
```   
see that in the 3rd call to ```echo``` it printed file names? that's because, echo never saw "src.*.js"